### DISCUSSION TITLE
`tty-test` Outputs "Process does not own terminal"

### DISCUSSION THREAD

**Discussion by ghost** [neovim/neovim#24341] [General]
### Problem

Hi everyone, I'm interested in helping with Neovim development. In order to familiarize myself with the codebase I decided to clone the repo, build Neovim, and run the tests.

A few tests are failing on my machine, and I'm trying to figure out if this is expected behavior or not.

The test I'm looking at specifically is the one in `test/functional/terminal/tui_spec.lua` called "paste: terminal mode".

[Test results](https://gist.github.com/schristy189/938b3b6bd860ac5a417c0b9fc7c2b8fe)

I investigated the issue and found that the test is spawning a program called `tty-test`, which fails with the message "Process does not own terminal".

If I compile this program manually with `gcc tty-test.c -luv`, I can see the same behavior outside of the Neovim test suite.

I thought that maybe this was an issue with my terminal setup, so I tested on several operating systems, terminals, and shells. Here are the results:

```
OS                   | Term           | tmux | Shell | Result                          | Notes
Arch rolling release | Alacritty      | Yes  | Fish  | "Process does not own terminal" |
Arch rolling release | Alacritty      | No   | Bash  | "Process does not own terminal" |
Arch rolling release | Alacritty      | No   | sh    | "Process does not own terminal" |
Arch rolling release | Gnome Terminal | No   | Fish  | "Process does not own terminal" |
Ubuntu 18.04.6 LTS   | urxvt          | No   | Bash  | "Process does not own terminal" |
Ubuntu 22.04.1 LTS   | Alacritty      | Yes  | Bash  | "Process does not own terminal" | Running in a Docker container
```

Does anyone else observe this behavior, and is this something the Neovim team would like me to investigate as a new contributor?

### Steps to reproduce

gcc test/functional/fixtures/tty-test.c -luv && ./a.out

### Expected behavior

Expected:

tty ready

Actual:

process does not own the terminal

### Neovim version (nvim -v)

NVIM v0.10.0-dev-0ce391086-dirty Build type: RelWithDebInfo LuaJIT 2.1.0-beta3

### Vim (not Nvim) behaves the same?

N/A

### Operating system/version

I use Arch BTW

### Terminal name/version

Alacritty 0.12.1

### $TERM environment variable

screen-256color

### Installation

Built from source

**Comment by clason** [comment]
Yes, that kind of failure is expected locally. The "source of truth" (and intended target) for these platform-specific tests are our CI (which run virtualized, so some bendiness is required for testing terminal-based user interactions).
